### PR TITLE
SUITES and kid (#151 #155 #156)

### PR DESCRIPTION
**Pull request by gselander**
Addressing #151

### CODE REVIEW

File: draft-ietf-lake-edhoc.md [hunk 1]
```diff
@@ -1110,20 +1110,20 @@ Error code 1 is used for errors that do not have a specific error code defined.
 Error code 2 MUST only be used in a response to message_1 in case the cipher suite selected by the Initiator is not supported by the Responder, or if the Responder supports a cipher suite more preferred by the Initiator than the selected cipher suite, see {{resp-proc-msg1}}. ERR_INFO is of type SUITES_R:
 
 ~~~~~~~~~~ CDDL
-SUITES_R : [ supported : 2* suite ] / suite
+SUITES_R : [ 2* suite ] / suite
 ~~~~~~~~~~~
 
-If the Responder does not support the selected cipher suite, then SUITES_R MUST include one or more supported cipher suites. If the Responder does not support the selected cipher suite, but supports another cipher suite in SUITES_I, then SUITES_R MUST include the first supported cipher suite in SUITES_I.
+If the Responder does not support the selected cipher suite, then SUITES_R MUST include one or more supported cipher suites. If the Responder does not support the selected cipher suite, but supports another cipher suite in SUITES_I, then SUITES_R MUST include the first supported cipher suite in SUITES_I. If the Responder does not support any cipher suite in SUITES_I, then it SHOULD include all its supported cipher suites in SUITES_R in any order.
```
Comment (marco-tiloca-sics) [inline draft-ietf-lake-edhoc.md:1116]: The paragraph before the CDDL notation mentions also one more error situation:

> or if the Responder supports a cipher suite more preferred by the Initiator than the selected cipher suite

The handling of this case seems not present in this paragraph, which focuses on the case where the Responder does not support the selected cipher suite. I suppose the missing sentence is something like:

"If the Responder supports the selected cipher suite but it also supports a cipher suite more preferred by the Initiator than the selected cipher suite, then SUITES_R MUST include such more preferred cipher suite".

Comment (marco-tiloca-sics) [inline draft-ietf-lake-edhoc.md:1116]: > then SUITES_R MUST include the first supported cipher suite in SUITES_I

This probably means to include **only** that cipher suite, with SUITE_R simply as an integer. Correct?

The same applies to my comment above for SUITE_R including "such more preferred cipher suite".

**Comment by gselander** [comment]
@marco-tiloca-sics Please check if your comments are addressed.

**Comment by marco-tiloca-sics** [comment]
> @marco-tiloca-sics Please check if your comments are addressed.

Yes, especially in [1]. Thanks!

[1] https://github.com/lake-wg/edhoc/pull/152/commits/1a44dfc8c2015278a3e4eb08399f0a153b8362c1

**Pull request merged**